Makefile

```diff
@@ -288,6 +288,19 @@ static-check: gocyclo
 
 get-deps: .get-deps-stamp
 
+DEP_VERSION=v0.4.1
+.PHONY: get-dep-linux
+get-dep-linux:
+	mkdir -p ./bin
+	curl -L https://github.com/golang/dep/releases/download/$(DEP_VERSION)/dep-linux-386 -o ./bin/dep
+	chmod +x ./bin/dep
+
+.PHONY: get-dep-osx
+get-dep-osx:
+	mkdir -p ./bin
+	curl -L https://github.com/golang/dep/releases/download/$(DEP_VERSION)/dep-darwin-386 -o ./bin/dep
```

Comment:
Is the only difference between `get-dep-linux` and `get-dep-osx` the inclusion of the software platform name in the source URL? If so, I'd suggest figuring out the platform at runtime and combining these into a single rule. Make the computer figure out the right source path rather than the developer.

Reply:
Consider something like:
```
PLATFORM:=$(shell uname -s)
ifeq (${PLATFORM},Linux)
	dep_arch=linux-386
else ifeq (${PLATFORM},Darwin)
	dep_arch=darwin-386
endif


.PHONY: get-dep
get-dep:
	mkdir -p ./bin
	curl -s -L https://github.com/golang/dep/releases/download/$(DEP_VERSION)/dep-${dep_arch} -o ./bin/dep
	chmod +x ./bin/dep
```

Are Linux and OSX the only software platforms we care about?

Also, the `clean` rule should probably clean up after this change. The idea is that `clean` should restore the repo to a pristine state.

This rule should also be idempotent. That is, running `make get-dep ; make get-dep` should only download the dep binary once. The ".stamp" stuff you see in some other rules is a common way to implement this.

Reply:
Yes, sorry, I missed the update to `clean`